### ISSUE TITLE
Fusiontree iterator with multiplicities

### ISSUE THREAD

**Issue by lkdvos** [Jutho/TensorKit.jl#192]
While updating SUNRepresentations to the latest version of TensorKit, I ran into an issue where it seems like the fusiontree iterator is generating trees that are not allowed. To reproduce this issue, running the tests for [this branch](https://github.com/QuantumKitHub/SUNRepresentations.jl/tree/tensorkitsectors) does the trick, but the following also showcases the problem (using that branch):

```julia
using SUNRepresentations, TensorKit

uncoupled = (Irrep[SU₃]((1, 0, 0)), Irrep[SU₃]((1, 1, 0)), Irrep[SU₃]((1, 0, 0)), Irrep[SU₃]((2, 1, 0)), Irrep[SU₃]((1, 0, 0)))
coupled = Irrep[SU₃]((6, 2, 0))
isdual = (false, true, true, false, true)

f = collect(fusiontrees(uncoupled, coupled, isdual))[2] # wrong vertices
Nsymbol(f.innerlines[end-1], f.uncoupled[end-1], f.innerlines[end]) # 1
f.vertices[end-1] # 2
```

I'm not 100% sure that the iterator is the culprit, but I checked with a couple versions of SUNRepresentations (specifically also before I changed the linear order) and the problem persists. The fusiontree iteration is definitely not something I am too familiar with though...

**Comment by Jutho** [comment]
Yes something is definitely wrong in the logic that deals with the vertices. I'll try to fix asap.

**Comment by Jutho** [comment]
Can you check if the branch [jh/fixfusiontreeiterator](https://github.com/Jutho/TensorKit.jl/tree/jh/fixfusiontreeiterator) fixes the problems? We should probably reenable SU(N) tests in the TensorKit testsuite.

**Comment by lkdvos** [comment]
This indeed seems to resolve the issues. I would suggest that we merge and tag this, then I'll tag the updated SUNRepresentations.jl version that depends only on TensorKitSectors.jl, which will make it easy to incorporate it in the tests here afterwards, without creating dependency cycles.
